### PR TITLE
fix(portal): Fix auth redirect and test client auth using acceptance tests

### PR DESCRIPTION
**Pull request by AndrewDryga**

### CODE REVIEW

File: elixir/apps/web/lib/web/auth.ex [hunk 1]
```diff
@@ -285,7 +288,7 @@ defmodule Web.Auth do
   Fetches the session token from the session and assigns the subject to the connection.
   """
   def fetch_subject(%Plug.Conn{} = conn, _opts) do
-    params = take_sign_in_params(conn.query_params)
+    params = take_sign_in_params(conn.params)
```
Comment (AndrewDryga) [inline elixir/apps/web/lib/web/auth.ex:291]: This is where the actual bug was.

File: elixir/apps/web/test/web/acceptance/auth/email_test.exs [hunk 2]
```diff
@@ -37,23 +37,110 @@ defmodule Web.Acceptance.SignIn.EmailTest do
...
+      )
+
+    # Sign In as an portal user
+    session
+    |> email_login_flow(account, identity.provider_identifier)
+    |> assert_el(Query.css("#user-menu-button"))
+    |> assert_path(~p"/#{account.slug}/sites")
+    |> Auth.assert_authenticated(identity)
+
+    # And then to a client
```
Comment (AndrewDryga) [inline elixir/apps/web/test/web/acceptance/auth/email_test.exs:110]: This is the test that will make sure it will never happen again.

**Review by jamilbk — APPROVED** [review]
LGTM! Nice work on getting the testing approach for this one

**Comment by jamilbk** [comment]
I hate to bring it up again, but still happening for me (and @bmanifold). But now it's only 50% of the time or so for me.

**Comment by jamilbk** [comment]
Maybe at this point it's just worth biting the bullet and adding records for `client-auth.firez.one` to scope the cookie properly?

**Comment by bmanifold** [comment]
Just as a data point, it seems to be happening for me 100% of the time.

**Pull request merged**